### PR TITLE
Make paginated calls to the Github GraphQL API

### PR DESCRIPTION
**Pull request by digitalronin**
This fixes a problem for teams who have more than 100 github repositories (e.g. the webops team, who have 110). 

The PR bot was only fetching open PRs for the first 100 repositories. This commit fixes that by using the pageInfo returned by the API to figure out when there are more repositories' PRs to fetch, and looping until all data has been retrieved.

This PR also splits the bot source code, separating the web server, PR fetcher, and output formatter into separate files. It adds a Dockerfile and instructions for running the bot locally, for development. This should also make it easier to move the bot into the cloud platform kubernetes cluster, when time permits.

I have tested this PR by briefly telling Heroku to deploy the bot from this branch, and then invoking it in a new Slack channel, and it works as expected.

### CODE REVIEW

File: lib/pr/fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,145 @@
+class PR
+  class Fetcher
+    HTTP = GraphQL::Client::HTTP.new("https://api.github.com/graphql") do
+      def headers(context)
+        unless token = context[:access_token] || ENV['GH_TOKEN']
+          fail "Missing GitHub access token"
+        end
+
+        {
+          "Authorization" => "Bearer #{ENV['GH_TOKEN']}"
```
Comment (SteveMarshall) [inline lib/pr/fetcher.rb]: You're setting a `token` variable a few lines before this; should you be using that here?

File: makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
+IMAGE := cloud-platform-slack-prbot
+VERSION := 1.0
+ORG := ministryofjustice
+TAGGED := $(ORG)/$(IMAGE):$(VERSION)
+
+.built-docker-image: Dockerfile Gemfile config.ru web.rb lib/pr.rb $(wildcard lib/pr/*.rb)
+	docker build -t $(IMAGE) .
+	docker tag $(IMAGE) $(TAGGED)
+	touch .built-docker-image
```
Comment (SteveMarshall) [inline makefile:16]: It'd be nice to have a corresponding `make clean` that removes the built docker image and the file you touch here.

**Review by SteveMarshall — APPROVED** [review]
This is really good, thanks! A couple of very minor suggestions, but I'm happy for you to fix those later if you'd rather.

I'd also have liked to have seen 947a814a broken up into a no-op “extract to classes” followed by a commit (or two?) adding the new functionality. As it is, it's a bit hard to see which bits are new and which bits are just moving around of existing, if you see what I mean? It's fine here, but worth bearing in mind for future.

File: lib/pr/fetcher.rb (hunk 1 above)
Comment (digitalronin) [reply lib/pr/fetcher.rb]: Good point. I missed that one.

File: makefile (hunk 2 above)
Comment (digitalronin) [reply makefile:16]: Good idea.

**Comment by digitalronin** [comment]
> I'd also have liked to have seen 947a814 broken up

Yes, that's exactly how I should have done it. I did a bunch of stuff while figuring out how it all worked, mixed in with pulling the structure apart (also as a part of figuring it out). 

I did think about disentangling this, to make the history clear, but I was already getting concerned about the amount of time I'd spent on this, so I decided to just push it as was.

**Pull request merged**